src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -186,6 +185,15 @@ private BufferSegment AllocateWriteHeadUnsynchronized(int count)
             return segment;
         }
 
+        private int GetSegmentSize(int sizeHint)
+        {
+            // First we need to handle case where hint is smaller then minimum segment size
+            var adjustedToMinimumSize = Math.Max(_minimumSegmentSize, sizeHint);
+            // After that adjust it to fit into pools max buffer size
+            var adjustedToMaximumSize = Math.Min(_pool.MaxBufferSize, adjustedToMinimumSize);
```

Comment:
The comment, variable name, and Math.Min/Max operations don't seem to align. Are we sure this is correct?

Reply:
Yep, `adjustedToMinimumSize` because of `_minimumSegmentSize`, `adjustedToMaximumSize` because of `_pool.MaxBufferSize`